wpimath/src/main/java/edu/wpi/first/math/struct/MatrixStruct.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.math.struct;
+
+import edu.wpi.first.math.MatBuilder;
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.Num;
+import edu.wpi.first.util.struct.Struct;
+import java.nio.ByteBuffer;
+
+public final class MatrixStruct<R extends Num, C extends Num> implements Struct<Matrix<R, C>> {
+  private final Nat<R> m_rows;
+  private final Nat<C> m_cols;
```

Comment:
I think these should be stored as int's, and getNum() called in the constructor.  And similarly for the other classes that do this, like Vector.

Reply:
We need to have access to the `Nat` instances for the call to `MatBuilder.fill()`. `VectorStruct` and `LinearStruct` can use int's, though.